.github/workflows/ci.yml

```
@@ -0,0 +1,309 @@
+            cargo check --verbose --all-features --manifest-path ./crates/${crate}/Cargo.toml;
+          done
+
+      - name: Check WASM
+        run: |
+          for crate in ${ALSO_WASM_CRATES}; do
+            cargo check --verbose --no-default-features --target wasm32-unknown-unknown \
+            --manifest-path ./crates/${crate}/Cargo.toml;
+          done        
+
+      - name: Check RISCV
+        env:
+          RUSTC_BOOTSTRAP:               "1"
+        run: |
+          for crate in ${ALSO_WASM_CRATES}; do
+            cargo check --verbose --no-default-features --target $RISCV_TARGET -Zbuild-std="core,alloc" \
+            --manifest-path ./crates/${crate}/Cargo.toml;
+          done       
+
+  dylint:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Dylint
+        run: |
+          cd linting/
+          # we are using a toolchain file in this directory
+          # add required components for CI
+          rustup component add rustfmt clippy
+          cargo check --verbose
+          cargo +nightly fmt --verbose --all -- --check
+          cargo clippy --verbose -- -D warnings;
+          # Needed until https://github.com/mozilla/sccache/issues/1000 is fixed.
+          unset RUSTC_WRAPPER
+          cargo test --verbose --all-features
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [check]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Build
+        run: |
+          ALL_CRATES="${PURELY_STD_CRATES} ${ALSO_WASM_CRATES}"
+          for crate in ${ALL_CRATES}; do
+            cargo build --verbose --all-features --release --manifest-path ./crates/${crate}/Cargo.toml;
+          done
+
+      - name: Build WASM
+        run: |
+          for crate in ${ALSO_WASM_CRATES}; do
+            cargo build --verbose --no-default-features --release --target wasm32-unknown-unknown \
+            --manifest-path ./crates/${crate}/Cargo.toml;
+          done
+
+  test:
+    runs-on: ubuntu-latest
+    needs: [check]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
```

```suggestion
      - name: Checkout
        uses: actions/checkout@v4
        with:
          fetch-depth: 100
          
      - name: Cache
        uses: Swatinem/rust-cache@v2
```

Setting up caching before sources checked out will not give the desired effect.